src/web/jed/echo.js

```diff
@@ -0,0 +1,114 @@
+import { byId } from './domQuery.js' 
+import { api } from './api.js'
+import { botDropdown } from './BotDropdown.js'
+
+class EchoApp {
+    constructor() {
+        this.queryButton = byId('query-echo-status')
+        this.queryButton.addEventListener('click', this.queryEchoStatus.bind(this))
+
+        this.startEchoButton = byId('echo-start-btn')
+        this.startEchoButton.addEventListener('click', this.startEcho.bind(this))
+
+        this.stopEchoButton = byId('echo-stop-btn')
+        this.stopEchoButton.addEventListener('click', this.stopEcho.bind(this))
+    }
+
+    updateStatus(status) {
+        // Update bounds, if the time is newer on this engineering status
+        const selected_bot_id = botDropdown.getSelectedBotId()
+        if (selected_bot_id == null) return
+
+        const thisBot = status.bots[botDropdown.getSelectedBotId()] 
+        if (thisBot == null)  return
+
+        const engineering_status = thisBot.engineering
+
+        if (engineering_status == null) return
+
+        if (engineering_status.echo == null) return
+
+        if (engineering_status.echo.echo_state == null) return
```

Comment:
I think we just need this line with a !== undefined (but it will need to be tested in a sandbox)